core/src/lib.rs

```diff
@@ -483,67 +589,84 @@ impl Game {
         }
         Ok(())
     }
+}
+
+// Implements logic for a game server without knowing about the transport layer.
+// Useful so that we can implement a real game server with web sockets, and the test
+// game server, and potentially a future peer to peer in-client server.
+pub struct GameServer {
```

Comment:
I had a similar comment whether this struct is going to be used outside of the server. But I'm starting to think your aim of this CL is to deconstruct the Game abstraction into three without bothering with moving the structs into different crates. Is this something you think we should do? Are there any downsides to doing it?

Reply:
The goal is for GameServer and GameAsPlayer to be the interfaces of Game that are consumed by the client and server respectively. That way the client and the server don't have to think much about the game nearly so much, they just need to hook up their corresponding interface properly.

That way, we can write a fairly end to end test here with a server and players that isn't bound to the client or server tech. Also, in the future we could run the server from _within_ one of the clients using WebRTC!

Reply:
What do you think about just breaking up core/src/lib.rs? It's gotten way too long for sure

Reply:
I'm just weary of leaving code common between client and server that is only specific to one or another. It's good as is, we can revisit later if necessary.